### ISSUE TITLE
Security annotations are not working using invokable controllers in the new way

### ISSUE THREAD

**Issue by BboyKeen** [symfony/symfony#16451]
Hi Symfony guys.

Looks like the security annotations are not working when using invokable controllers in the new way, i.e  like in this PR https://github.com/symfony/symfony/pull/11193

Here is the skeleton of my class : 

```php
class Create
{
    /**
     * @Security("has_role('ROLE_CORPORATE')")
     */
    public function __invoke(Request $request)
    {
     .....
    }
}
```

In the old way, the security annotation is used and triggers exception correctly
```php
create-program:
    path:     /programs/new
    defaults: { _controller:  acme.program.controller.command.create:__invoke }
```

In the new way, nothing happens and all roles can access the page
```php
create-program:
    path:     /programs/new
    defaults: { _controller:  acme.program.controller.command.create }
```

**Comment by iltar** [comment]
Just a guess, but what about the `@Route` will that work? What happens if you put them on your class level annotations?

**Comment by ogizanagi** [comment]
That's an issue in the `SensioFrameworkExtraBundle`, more precisely in the [`ControllerListener`](https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/master/EventListener/ControllerListener.php#L59) when trying to get the method name from the controller name. The same would happen for any FrameworkExtra annotation.

**Comment by BboyKeen** [comment]
@iltar The Route annotation is not working either.
 The same happens when I put them on class level.

**Comment by BboyKeen** [comment]
@ogizanagi I think you're right. I can reopen it in the `SensioFrameworkExtraBundle` repository.

**Comment by xabbuh** [comment]
@BboyKeen You can do that to keep track of the issue. But we also already have PR.

**Comment by lunetics** [comment]
Is that PR already merged? stumbled onto the same problem

**Comment by ogizanagi** [comment]
@lunetics : It should have been fixed in https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/398, which is merged and part of the 3.0.16 release of the bundle.
